### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by ruimonteiro93**
Implementation of moxy's react-icon base component.

### CODE REVIEW

File: demo/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+node_modules
+.next/
+out/
+dist
```
Comment (paulobmarcos) [inline demo/.gitignore:4]: Demo won't have dist folder, we can remove this :+1:

File: demo/shared/modules/icons/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React, { forwardRef } from 'react';
```
Comment (paulobmarcos) [inline demo/shared/modules/icons/index.js:1]: The path to this file should be: `demo/shared/modules/react-icons/index.js`

File: package.json [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
-    "es"
+    "es",
+    "dist"
   ],
-  "homepage": "https://github.com/moxystudio/react-lib-template#readme",
+  "homepage": "https://github.com/moxystudio/react-icon#readme",
   "author": "Rui Monteiro <rui.monteiro@moxy.studio>",
   "license": "MIT",
   "keywords": [
     "template"
```
Comment (paulobmarcos) [inline package.json:16]: Same with keywords here.

File: package.json [hunk 4]
```diff
@@ -1,30 +1,32 @@
 {
-  "name": "@moxy/react-lib-template",
+  "name": "@moxy/react-icon",
   "version": "0.0.0",
   "description": "A template that aims to make the implementation of React component packages easier and more methodic.",
```
Comment (paulobmarcos) [inline package.json:4]: Please give a better description here inline with the package.

File: README.md [hunk 5]
```diff
@@ -1,15 +1,113 @@
...
+
+The contents of the `SVG` that should be rendered. 
+In the case of the prop being an object, it must be the `Promise` that gets the contents of the `SVG`.
+
+#### className
+
+Type: `string` | Required: `false`
+
+A className to apply to the component.
+
```
Comment (paulobmarcos) [inline README.md:85]: Delete Extra Line

File: README.md [hunk 6]
```diff
@@ -1,15 +1,113 @@
...
+[build-status-url]:https://github.com/moxystudio/react-icon/actions
+[build-status-image]:https://img.shields.io/github/workflow/status/moxystudio/react-icon/Node%20CI/master
+[codecov-url]:https://codecov.io/gh/moxystudio/react-icon
+[codecov-image]:https://img.shields.io/codecov/c/github/moxystudio/react-icon/master.svg
+[david-dm-url]:https://david-dm.org/moxystudio/react-icon
+[david-dm-image]:https://img.shields.io/david/moxystudio/react-icon.svg
+[david-dm-dev-url]:https://david-dm.org/moxystudio/react-icon?type=dev
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/react-icon.svg
+
+A component that renders the contents of an Icon.
```
Comment (paulobmarcos) [inline README.md:17]:
```suggestion
A component to render svg icons.
```

File: README.md [hunk 7]
```diff
@@ -1,15 +1,113 @@
...
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+This package was implemented to help implement an `SVG` icon on your projects.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+Create an `icons` folder on your `shared/modules`, and export icons from there.
```
Comment (paulobmarcos) [inline README.md:33]:
```suggestion
Create a `react-icons` folder on your `shared/modules`, and export icons from there.
```

File: README.md [hunk 8]
```diff
@@ -1,15 +1,113 @@
...
+
+import React, { forwardRef } from 'react';
+import Icon from '@moxy/react-icon';
+
+const arrowLeftSvg = import(/* webpackChunkName: "svg-sprite" */ '../media/arrow-left.inline.svg');
+const ArrowLeftIcon = forwardRef((props, ref) => <Icon ref={ ref } { ...props } svg={ arrowLeftSvg } />);
+
+export { ArrowLeftIcon };
+
+export default Icon;
```
Comment (paulobmarcos) [inline README.md:46]: I think there is no need to export icon here.

File: README.md [hunk 9]
```diff
@@ -1,15 +1,113 @@
...
+
+This package was implemented to help implement an `SVG` icon on your projects.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+Create an `icons` folder on your `shared/modules`, and export icons from there.
+
+```js
+// shared/modules/icons/index.js
```
Comment (paulobmarcos) [inline README.md:36]: The path to this file should be: shared/modules/react-icons/index.js

File: README.md [hunk 10]
```diff
@@ -1,15 +1,113 @@
...
+```js
+import { ArrowLeftIcon } from '../..//shared/modules';
+
+// (...)
+
+render() {
+    <div>
+        <ArrowLeftIcon className={ styles.arrowLeft } />
+    </div>
+}
```
Comment (paulobmarcos) [inline README.md:60]: I would prefer to have a better example:

```js
import { ArrowLeftIcon } from '../shared/modules/react-icons';

const MyComponent = ({ children }) => (
    <div>
        { children }
        <ArrowLeftIcon />
    </div>
);
```

File: README.md [hunk 11]
```diff
@@ -1,15 +1,113 @@
...
+```
+
+Also, don't forget to export the icons on your `shared/modules/index.js`. Just add this line:
```
Comment (paulobmarcos) [inline README.md:63]: There is no `index.js` file in modules anymore. We can remove this suggestion

File: README.md [hunk 12]
```diff
@@ -1,15 +1,113 @@
...
+```sh
+$ npm test
+$ npm test -- --watch # during development
+```
+
+## Demo
+
+A demo [Next.js](https://nextjs.org/) project is available in the [`/demo`](./demo) folder so you can try out this component.
+
+First, build the `{react-icon}` project with:
```
Comment (paulobmarcos) [inline README.md:98]:
```suggestion
First, build the `react-icon` project with:
```

File: README.md [hunk 13]
```diff
@@ -1,15 +1,113 @@
...
+
+```sh
+$ npm install @moxy/react-icon
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+This package was implemented to help implement an `SVG` icon on your projects.
```
Comment (paulobmarcos) [inline README.md:29]:
```suggestion
This package was implemented to help import `SVG` icons in react projects.
```

File: README.md [hunk 14]
```diff
@@ -1,15 +1,103 @@
...
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+This package was implemented to help import `SVG` icons in `react` projects.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+Create an `react-icons` folder on your `shared/modules`, and export icons from there.
```
Comment (fsdiogo) [inline README.md:33]:
```suggestion
Create a `react-icons` folder on your `shared/modules`, where you'll export your icons.
```

File: README.md [hunk 15]
```diff
@@ -1,15 +1,103 @@
...
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+This package was implemented to help import `SVG` icons in `react` projects.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+Create a `react-icons` folder on your `shared/modules`, and export icons from there.
```
Comment (paulobmarcos) [inline README.md:33]: Now that I'm thinking about it... Should this be the example? This seems very related with our `next-with-moxy` 🤔 

Should we have an example that would work in almost any react project?

File: README.md [hunk 16]
```diff
@@ -1,15 +1,91 @@
...
+This package was implemented to help import `SVG` icons in `react` projects.
 
 ## Usage
 
-This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+```js
+import React, { forwardRef } from 'react';
+import Icon from '@moxy/react-icon';
+
+const arrowLeftSvg = import(/* webpackChunkName: "svg-sprite" */ '../media/arrow-left.inline.svg');
```
Comment (paulobmarcos) [inline README.md:37]:
```suggestion
const arrowLeftSvg = import('../media/arrow-left.inline.svg');
```

File: demo/shared/media/twitter.inline.svg [hunk 17]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (nlfonseca) [inline demo/shared/media/twitter.inline.svg:1]: I suggest optimizing this icon to show the best practice
https://jakearchibald.github.io/svgomg/

File: src/Icon.js [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+import classNames from 'classnames';
+import Svg from '@moxy/react-svg';
+
+export const Icon = forwardRef(({ className, ...rest }, ref) => {
+    const finalProps = {
+        ...rest,
+        className: classNames('react-icon', className),
+    };
+
+    return <Svg ref={ ref } { ...finalProps } />;
```
Comment (dominguesgm) [inline src/Icon.js:12]: I may be wrong, but I think you still have to pass the ref as `forwardedRef`, as it is the prop the `Svg` component receives.

**Pull request merged**